handlers_test.go

```diff
@@ -40,20 +59,177 @@ func (suite *HandlerTestSuite) TestTopicListAll() {
+	router.HandleFunc("/v1/projects/{project}/topics", WrapConfig(TopicListAll, cfgKafka, &brk))
+	router.ServeHTTP(w, req)
 	suite.Equal(200, w.Code)
 	suite.Equal(expResp, w.Body.String())
 }
 
+func (suite *HandlerTestSuite) TestPublish() {
+	//broker.Initialize(kafkaCfg.Server)
+	postJSON := `{
+  "messages": [
+    {
+      "attributes": [
+        {
+          "key": "foo",
+          "value": "bar"
+        }
+      ],
+      "data": "YmFzZTY0ZW5jb2RlZA=="
+    }
+  ]
+}`
+	url := "http://localhost:8080/v1/projects/ARGO/topics/mocktopic:publish"
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(postJSON)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	expJSON := `{
+   "messageIDs": [
+      "1"
+   ]
+}`
+
+	cfgKafka := config.NewKafkaCfg()
+	cfgKafka.LoadStrJSON(suite.cfgStr)
+	brk := brokers.MockBroker{}
+	router := mux.NewRouter().StrictSlash(true)
+	w := httptest.NewRecorder()
+	router.HandleFunc("/v1/projects/{project}/topics/{topic}:publish", WrapConfig(TopicPublish, cfgKafka, &brk))
+	router.ServeHTTP(w, req)
+	suite.Equal(200, w.Code)
+	suite.Equal(expJSON, w.Body.String())
+
+}
+
+func (suite *HandlerTestSuite) TestPublishMultiple() {
+	//broker.Initialize(kafkaCfg.Server)
+	postJSON := `{
+  "messages": [
+    {
+      "attributes": [
+        {
+          "key": "foo",
+          "value": "bar"
+        }
+      ],
+      "data": "YmFzZTY0ZW5jb2RlZA=="
+    },
+    {
+      "attributes": [
+        {
+          "key": "foo2",
+          "value": "bar2"
+        }
+      ],
+      "data": "YmFzZTY0ZW5jb2RlZA=="
+    },
+    {
+      "attributes": [
+        {
+          "key": "foo2",
+          "value": "bar2"
+        }
+      ],
+      "data": "YmFzZTY0ZW5jb2RlZA=="
+    }
+  ]
+}`
+	url := "http://localhost:8080/v1/projects/ARGO/topics/mocktopic:publish"
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(postJSON)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	expJSON := `{
+   "messageIDs": [
+      "1",
+      "2",
+      "3"
+   ]
+}`
+
+	cfgKafka := config.NewKafkaCfg()
+	cfgKafka.LoadStrJSON(suite.cfgStr)
+	brk := brokers.MockBroker{}
+	router := mux.NewRouter().StrictSlash(true)
+	w := httptest.NewRecorder()
+	router.HandleFunc("/v1/projects/{project}/topics/{topic}:publish", WrapConfig(TopicPublish, cfgKafka, &brk))
+	router.ServeHTTP(w, req)
+	suite.Equal(200, w.Code)
+	suite.Equal(expJSON, w.Body.String())
+
+}
+
+func (suite *HandlerTestSuite) TestPublishError() {
+	//broker.Initialize(kafkaCfg.Server)
+	postJSON := `{
+  "messages": [
+    {
+      "attributes": [
+        {
+          "key": "foo",
+          "valu2RlZA=="
+    },
+    {
+      "attributes": [
+        {
+          "key": "foo2",
+          "value": "bar2"
+        }
+      ],
+      "data": "YmFzZTY0ZW5jb2RlZA=="
+    }
+  ]
+}`
+	url := "http://localhost:8080/v1/projects/ARGO/topics/mocktopic:publish"
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(postJSON)))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
@kaggis I am not sure but maybe it's better to remove this conditional. The point of the test is to catch a correct response in case of a malformed json, so maybe there is no need to log it anyhow.